snc.sh

```diff
@@ -227,31 +227,57 @@ function create_pvs() {
 # This follows https://blog.openshift.com/enabling-openshift-4-clusters-to-stop-and-resume-cluster-vms/
 # in order to trigger regeneration of the initial 24h certs the installer created on the cluster
 function renew_certificates() {
-    # Get the cli image from release payload and update it to bootstrap-cred-manager resource
-    cli_image=$(${OC} adm release -a ${OPENSHIFT_PULL_SECRET_PATH} info ${OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE} --image-for=cli)
-    ${YQ} write kubelet-bootstrap-cred-manager-ds.yaml.in spec.template.spec.containers[0].image ${cli_image} >kubelet-bootstrap-cred-manager-ds.yaml
+    # Create the unsupported cert rotation config for a day (validity will increase by 30 times so it will become 30days)
+    ${OC} create -n openshift-config configmap unsupported-cert-rotation-config --from-literal='base=86400s'
 
-    ${OC} apply -f kubelet-bootstrap-cred-manager-ds.yaml
-    rm kubelet-bootstrap-cred-manager-ds.yaml
+    # Delete the pods to have cert rotation
+    ${OC} delete pods --all -A --force --grace-period=0
 
-    # Delete the current csr signer to get new request.
-    ${OC} delete secrets/csr-signer-signer secrets/csr-signer -n openshift-kube-controller-manager-operator
+    # Wait till the api server again started
+     while ! ${OC} get nodes >/dev/null 2>&1; do
+         sleep 3
+     done
+     echo "API server is up"
+
+    # Force the rotation for secrets
+    ${OC} get secret -A -o json | ${JQ} -r '.items[] | select(.metadata.annotations."auth.openshift.io/certificate-not-after" | .!=null and fromdateiso8601<='$( date --date='+1year' +%s )') | "-n \(.metadata.namespace) \(.metadata.name)"' | \
+        xargs -n3 ${OC} patch secret -p='{"metadata": {"annotations": {"auth.openshift.io/certificate-not-after": null}}}'
 
-    # Wait for 5 min to make sure cluster is stable again.
+    # Delete the current csr signer to get new request.
+    ${OC} delete secrets/csr-signer-signer secrets/csr-signer -n openshift-kube-controller-manager-operator || true
```

Comment:
These secrets are also affected by the action just before. Maybe it is not needed?